### PR TITLE
🦺(models) make ProfileActivity presence mandatory

### PR DESCRIPTION
**Pull request by SergioSim**
## Purpose

The latest version of the xapi video profile (1.0.3) seems to enforce the presence of a video profile activity in the
`context.contextActivities.category` field, which was previously validated to optionally contain the profile activity value.

## Proposal

- [x] make the presence of a ProfileActivity mandatory in video and virtual classroom profile statements

### CODE REVIEW

File: src/ralph/models/xapi/video/contexts.py [hunk 1]
```diff
@@ -32,17 +31,13 @@
...
-    property.
+    """Pydantic model for video `context`.`contextActivities`.`category` property.
 
     Attributes:
         id (str): Consists of the value `https://w3id.org/xapi/video`.
-        definition (dict): see ProfileActivity.
     """
 
     id: Literal["https://w3id.org/xapi/video"] = "https://w3id.org/xapi/video"
-    definition: ProfileActivity
```
Comment (quitterie-lcs) [inline src/ralph/models/xapi/video/contexts.py:45]: You're right, it is not a mandatory field to have for validation. But we can keep it as optional, no? WDYT?

**Review by cyrillay — APPROVED** [review]

**Review by wilbrdt — APPROVED** [review]
LGMT

File: src/ralph/models/xapi/video/contexts.py [hunk 2]
```diff
@@ -31,32 +25,36 @@
...
 
     category: Union[
-        VideoContextContextActivitiesCategory,
-        List[VideoContextContextActivitiesCategory],
+        VideoProfileActivity, List[Union[VideoProfileActivity, BaseXapiActivity]]
     ]
 
+    @validator("category")
+    @classmethod
+    def check_presence_of_profile_activity_categoy(
```
Comment (quitterie-lcs) [inline src/ralph/models/xapi/video/contexts.py:41]:
```suggestion
    def check_presence_of_profile_activity_category(
```

**Review by quitterie-lcs — COMMENTED** [review]
Some remarks and we are ready to go !

File: src/ralph/models/xapi/video/contexts.py (hunk 2 above)
Comment (SergioSim) [reply src/ralph/models/xapi/video/contexts.py]: Well Spotted) Thanks)

File: src/ralph/models/xapi/video/contexts.py [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
-
 from uuid import UUID
 
-from pydantic import Field, NonNegativeFloat, PositiveInt
+from pydantic import Field, NonNegativeFloat, PositiveInt, validator
 
 from ..base.contexts import BaseXapiContext, BaseXapiContextContextActivities
 from ..base.unnested_objects import BaseXapiActivity
-from ..concepts.activity_types.scorm_profile import ProfileActivity
+from ..concepts.activity_types.video import VideoProfileActivity
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply src/ralph/models/xapi/video/contexts.py]: Good point) You are right; indeed, they are not reusable) updated this part)

**Review by quitterie-lcs — APPROVED** [review]

**Pull request merged**